### ISSUE TITLE
[a11y Support] MHV Secure Messaging: NVDA behavior on <va-modal>

### ISSUE THREAD

**Issue by oleksii-morgun-va** [department-of-veterans-affairs/va.gov-team#61387]
## About your team

- Team name: MHV Secure Messaging
-	OCTO-DE product owner: Patricia Henry
-	Product manager: Lichelle Bain
-	Designer: Joel Calumpong 
-	FE engineer: Oleksii Morgun @oleksii-morgun-va 
- Accessibility specialist:
- Content writer: 
-	Product/team Slack channel:
 
### What does your team need support for? Check all that apply.
- [ ] Launching one or more unauthenticated pages, online forms, apps, or tools on VA.gov
- [ ] Making revisions to one or more existing VA.gov pages
- [ ] Accessibility review of wireframes and/or prototypes
- [x] Accessibility issue (general request)
- [ ] Accessibility PDF review request
- [ ] Accessibility user research support ([submit a separate, additional request](https://github.com/department-of-veterans-affairs/va.gov-team/issues/new?assignees=coforma-terry%2Cangelafowler82&labels=508%2FAccessibility%2C+sitewide+CAIA%2C+sitewide+accessibility%2C+CAIA-a11y-research&template=caia-a11y-research.yaml&title=%5BCAIA+A11y+Research%5D%3A+Team+name%2C+product+name))
- [ ] DST component or pattern 
- [ ] Something else 
 
### Give a brief description of what your team needs support with:

My team needs support with understanding the behavior of NVDA screen reader on <va-modal> open. More details in [this slack thread](https://dsva.slack.com/archives/C8E985R32/p1688131671037029)

 **_In our use case, this can lead to an irreversible action if a user mistakenly executes a primary button, thinking they are clicking on an "X" to close a modal._**

#### Oleksii 

- When opening a <va-modal> that contains primary and secondary buttons, NVDA reads through entire modal content starting from the title. 
    - Focus is set to a "X" button as announced to a screen reader user. 
- However, as it announces through the content and reaches buttons below (**visually focus is not moving though**) and if a user happens to hit Enter key, **NVDA will actually fire an event of the primary button, not a "X" button where the focus is set to at that moment.** 
- I am not very familiar with the settings of the tool. It may be a configuration issue or a defect.
    -  Attaching a video for demonstration. Thank you in advance!
- _**VoiceOver or Windows Narrator will execute closing a modal in this scenario, although those tools also continue reading through the modal in the same way. I am guessing it is NVDA specific issue.**_
    - But wanted to make sure we are not missing any config in NVDA settings that would force focus to be moved as content is announced through.

https://github.com/department-of-veterans-affairs/va.gov-team/assets/124186314/20fbc19d-53bd-485e-8dbe-a41b99b0a492

### Steps to Recreate this: 
https://design.va.gov/storybook/iframe.html?id=components-va-modal--default&viewMode=docs#warning
_Comments and video added by @sara-amanda from Slack thread._ 

### Will this new product be released incrementally (for instance 25% of users initially)? 
  - [ ] Yes
  - [ ] No
 
**Note:** If you checked yes, we'll reach out to discuss details about the content in the react widget.
  
### When do you expect to launch your product to 100% of users?
  
*Please provide an estimated date so our team can create other relevant tickets.*

### When does this work need to be completed?
- Estimated launch date:
- Estimated staging review date:
- Content, Accessibility, and IA work needed by: 

### Will this work be going through the Collaboration Cycle?
- [ ] Yes 
- [x] No

### If no ...
- Skip to the _supporting artifacts_ section, below.
 
#### Please provide the link to your Collaboration Cycle ticket: 
- Ticket (number): 
- Collab. Cycle Ticket Opened (date): 
 
#### Please check all of the phases in the Collaboration Cycle you have completed: 
- [ ] Design Intent
- [ ] Midpoint Review
- [ ] Staging
 
##### Design Intent
- [ ] Accessibility
- [ ] Content
- [ ] IA
- [ ] Translation
- [ ] Already completed, no assistance needed.
- [ ] Unknown at this time.
##### Midpoint
- [ ] Accessibility
- [ ] Content
- [ ] IA
- [ ] Translation
- [ ] Already completed, no assistance needed.
- [ ] Unknown at this time.
##### Staging
- [ ] Accessibility
- [ ] Content
- [ ] IA
- [ ] Translation
- [ ] Already completed, no assistance needed.
- [ ] Unknown at this time.
  
## Supporting artifacts
*Please provide supporting artifacts as available.*

- Link to product outline:
- Link to or attach user flows:
- Link to prototype or mockups:
- Link to your research plan: 
- Link to any epics/issues that may be helpful:

#### Do you plan to bring this to an upcoming content office hours session?
- [ ] Yes, benefit content office hours (Thursdays, 3:00 p.m. to 3:45 p.m. ET)
- [ ] Yes, health  content office hours (Thursdays, 11:00 a.m. to 11:30 a.m. ET)
- [ ] Yes, unauth office hours (Mondays, 11:00 a.m. to 11:30 a.m. ET)
- [ ] No, but I'd like to schedule time to talk about this request
- [ ] No, let's work asynchronously and meet if needed

 **Note:** If we think this work would benefit from a collaborative session with you, we may ask you to bring it to office hours or set up a separate time to meet.
 
### Accessibility Help in Slack
The [#accessibility-help channel](https://dsva.slack.com/archives/C8E985R32) in the DSVA Slack, is also available to you as a resource. Everyone is welcome to ask questions or get help from our a11y (accessibility) specialists. Tag `@Terry Nichols` to get a11y help asap.

## Next steps

**Once you’ve submitted this ticket, please post a link to this issue in the [#sitewide-content-ia](https://dsva.slack.com/channels/sitewide-content-ia) Slack channel and tag `@Terry Nichols`.**

- If you also need engineering support from the Public Websites team, fill out their [intake request form](https://github.com/department-of-veterans-affairs/va.gov-team/issues/new?assignees=Public+Websites%2C+Sitewide+content&labels=vsa-public-websites%2C+vsa%2C+vsa-public-websites-intake%2C+sitewide-content%2C+needs-grooming&template=public-websites-intake.md&title=%3CType+of+Request%3E+from+%3CTeam%3E).
  
- If you need a page/URL redirected, a URL changed or a vanity URL set up, please submit a [Redirect, URL change, or vanity URL request](https://github.com/department-of-veterans-affairs/va.gov-team/issues/new?assignees=mnorthuis&labels=ia&template=redirect-request.md&title=Redirect+Request)
 
 - If you are creating an [experimental design,](https://design.va.gov/about/contributing-to-the-design-system/experimental-components-and-patterns#what-is-an-experimental-component-or-pattern?) please [also contact the DST. ](https://design.va.gov/about/contributing-to-the-design-system/suggest-an-addition-or-update)
- If you are requesting [a11y support for user research,](https://github.com/department-of-veterans-affairs/va.gov-team/issues/new?assignees=coforma-terry%2Cangelafowler82&labels=508%2FAccessibility%2C+sitewide+CAIA%2C+sitewide+accessibility%2C+CAIA-a11y-research&template=caia-a11y-research.yaml&title=%5BCAIA+A11y+Research%5D%3A+Team+name%2C+product+name) please complete this form, next.

**Comment by sara-amanda** [comment]
## CAIA Update
### Adding Similar Behavior Notes
[Per Laura](https://dsva.slack.com/archives/C8E985R32/p1688133077846809?thread_ts=1688131671.037029&cid=C8E985R32) @laflannery 
- The Sign in modal on the VA site acts the same way with NVDA. 
- The focus is on the X but when I hit enter the [Login.gov](http://login.gov/) button was activated.  
- And it's also worth noting that the Sign in modal is not using the modal component from the DST

**Comment by sara-amanda** [comment]
## CAIA Update

- The new DST bug ticket is being submitted today 7/20/2023 after discussing further in the a11y specialists meeting today, and with @jeana-adhoc : [Bug] DST: MHV Secure Messaging: SR behavior on warning <va-modal> #62017. 
- Please follow that ticket @oleksii-morgun-va for future updates on this bug. 
- Closing this ticket @coforma-terry
